src/main/java/com/exasol/adapter/dynamodb/mapping/TableMappingDefinition.java

```diff
@@ -72,19 +96,22 @@ public String getExasolName() {
      *         list or map from DynamoDB
      */
     public boolean isRootTable() {
-        return this.pathToNestedTable == null;
+        return this.pathToNestedTable.size() == 0;
```

Comment:
Have you checked that this field can't be null in your implementation? Because if you call size() on the null you get NullPointer. If you are trying to avoid nulls in this way, that's not what I meant

Reply:
Yes, I chnaged my implementation so that alwys a path is assigned. As my path does support empty paths I though this might be more elegant than using an Optional.